src/components/calcite-switch/calcite-switch.scss

```diff
@@ -84,14 +84,15 @@
 :host(:hover),
 :host(:focus) {
   .handle {
-    @apply border-color-brand shadow-outline-active;
+    @apply border-color-brand-hover;
+    box-shadow: inset 0 0 0 1px var(--calcite-ui-brand-hover);
     @apply right-auto;
```

Comment:
It also looks like `@apply right-auto;` is adding a wobble effect to the handle on :hover when dir is "rtl". Can you try removing to see if it fixes that issue?